### ISSUE TITLE
[1.7] Average server load going up 2-3 times after upgrade

### ISSUE THREAD

**Issue by schlaegerz** [meteor/meteor#9949]
I'm not sure if I have enough information to post a full bug, but after upgrading my production server to 1.7 and switchin to dynamic imports my average server load went up by a factor of 2-3 times. I expected a little bit more but doubling seems like a bug or a really poor implementation by me. 

The number of users on my site did not change by any significant amount.

Is this an expected amount of efficiency hit?
![2018-06-05 2](https://user-images.githubusercontent.com/8098360/41009068-9948aa9c-68e3-11e8-8beb-6efda37f609d.png)

**Comment by benjamn** [comment]
No, that's definitely not expected. Could be related to the Mongo 3.6 update and/or the `mongodb@3.0.10` Node driver update?

@veered Do you have any tricks for profiling in production? (Long shot, but I had to ask)

**Comment by veered** [comment]
Yep, open a meteor shell in production using [qualia:prod-shell](https://github.com/qualialabs/prod-shell) and then you can trigger a profile from the meteor shell using [qualia:profile](https://github.com/qualialabs/profile) (as described in the "Runtime Profiling" section of the qualia:profile README). 

This will write a `*.cpuprofile` file to disk on your server, and this file can be opened in the Chrome Javascript CPU profiler.

We do prod CPU profiles all the time at Qualia!

**Comment by veered** [comment]
If you don't have an easy way of transferring files from your server to your local computer, `scp` might be a good choice: 

https://stackoverflow.com/questions/14325956/using-scp-in-terminal

**Comment by schlaegerz** [comment]
Thanks I'll look into it tomorrow, I have a feeling it is related to the mongo driver, most of my queries seem to be taking longer than expected and I am having mongo issues

**Comment by schlaegerz** [comment]
Also of note is my memory is growing really fast

**Comment by schlaegerz** [comment]
I tried using those two packages, but the profiler is not writing anything into the file. The meteor shell doesn't seem to have the right permissions to write the file. 

I can add and edit files from the ssh prompt so its just the meteor shell/ this tool. 

Do you have any suggestions on how to fix that?

**Comment by schlaegerz** [comment]
I am also having this [issue](https://github.com/meteor/meteor/issues/9944) but I have these performance problems well before any crash occurs

**Comment by veered** [comment]
@schlaegerz Would you mind uploading the full profile?

**Comment by schlaegerz** [comment]
@veered github doesn't seem to let me upload


[Here is a minute long profile](https://github.com/meteor/meteor/files/2077306/myprofile_large.cpuprofile.zip)
[And a 10 second one](https://github.com/meteor/meteor/files/2077307/myprofile.cpuprofile.zip)

**Comment by veered** [comment]
Eyeballing the profile, I bet you have a bunch of polling observes. 

If you use [qualia:analyze-observes](https://github.com/qualialabs/analyze-observes) in production from the Meteor shell, it'll give you a full list of all active observers, and in particular it'll report any polling observers & the reason why they are polling.

Using observers in polling mode is almost always a total performance disaster.

**Comment by veered** [comment]
Also, per usual, the BSON parser is incredibly slow. Sigh. I tried the native BSON parser a while back, and it's actually SLOWER than the BSON parser implemented in JS.

**Comment by schlaegerz** [comment]
I am working on getting it be a better view of the observers, but nothing in it looks out of place, I have a whole bunch of them, but that's because I have a bunch of users doing a bunch of different things. I'm not sure why this would have regressed.

Is how do I go back to a javascript BSON parser? If that is the issue.

**Comment by veered** [comment]
How did you look at your observers? Kadira doesn't really provide a complete view.

You are using the JS BSON parser; I mentioned the native parser because I tried it out in the hopes that it was faster than the JS parser but it is not.

**Comment by schlaegerz** [comment]
This ended up being unrelated to core meteor changes. 

In an effort to move to more dynamic loading to take advantage of Meteor 1.7 I had moved some code around incorrectly. In the end I ended up in a template essentially this:

```
onCreated(function(){
   this.autorun(()=>{
       Meteor.subscribe('item', ()=>{
             Meteor.call('viewItem', item._id)
       })
   })
}
```

View item would change the subscription, which would make it call the method again. 

Thank you so much for your help profiling this, it really did help me figure this out, and I fixed a whole lot of smaller performance issues because of this. We really should point to these packages somewhere in the meteor documentation as they are super helpful and showed me things I would never have seen otherwise.
